src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/DirectoryBasedTemplate.cs

```diff
@@ -0,0 +1,412 @@
+                }
+
+                IFile? partialConfigFile = primarySourceConfig.Parent?.EnumerateFiles(partialConfigFileName, SearchOption.TopDirectoryOnly).FirstOrDefault(x => string.Equals(x.Name, partialConfigFileName));
+
+                if (partialConfigFile == null)
+                {
+                    throw new TemplateAuthoringException(string.Format(LocalizableStrings.SimpleConfigModel_AuthoringException_MergeConfiguration_FileNotFound, partialConfigFileName), partialConfigFileName);
+                }
+
+                JObject partialConfigJson = partialConfigFile.ReadJObjectFromIFile();
+                combinedSource.Merge(partialConfigJson);
+            }
+
+            return combinedSource;
+        }
+
+        /// <summary>
+        /// Checks if the template is supported by current generator version.
+        /// </summary>
+        /// <exception cref="NotSupportedException">when the template is not supported by current generator version.</exception>
+        /// <exception cref="InvalidOperationException">when check for the version leads to unexpected result.</exception>
+        private void CheckGeneratorVersionRequiredByTemplate()
+        {
+            if (string.IsNullOrWhiteSpace(ConfigModel.GeneratorVersions))
+            {
+                return;
+            }
+
+            string allowedGeneratorVersions = ConfigModel.GeneratorVersions!;
+
+            if (!VersionStringHelpers.TryParseVersionSpecification(allowedGeneratorVersions, out IVersionSpecification? versionChecker))
+            {
+                throw new NotSupportedException(string.Format(LocalizableStrings.RunnableProjectGenerator_Exception_TemplateVersionNotSupported, allowedGeneratorVersions, RunnableProjectGenerator.GeneratorVersion));
+            }
+
+            if (versionChecker is null)
+            {
+                throw new InvalidOperationException($"{nameof(versionChecker)} cannot be null when {nameof(VersionStringHelpers.TryParseVersionSpecification)} is 'true'");
+            }
+
+            if (!versionChecker.CheckIfVersionIsValid(RunnableProjectGenerator.GeneratorVersion))
+            {
+                throw new NotSupportedException(string.Format(LocalizableStrings.RunnableProjectGenerator_Exception_TemplateVersionNotSupported, allowedGeneratorVersions, RunnableProjectGenerator.GeneratorVersion));
+            }
+        }
+
+        /// <summary>
+        /// The method validates loaded cpnfiguration. The errors and warnings are printed using logger.
+        /// The warning messages is mainly only for template authoring, and should be used for template validation, so for now they are logged in debug level only.
+        /// https://github.com/dotnet/templating/issues/2623.
+        /// </summary>
+        /// <exception cref="TemplateValidationException">in case of validation fails.</exception>
+        private void PerformTemplateValidation()
```

Comment:
Should this actually return list of `IValidationEntry`? Similarly with other checking methods in this template type